<launch>
    <node name = "aruco_pose_pub" pkg = "am_aruco_tracker" type = "aruco_pose_pub" output = "screen">
        <!-- publish rate -->
        <rosparam param="publish_rate">10</rosparam>
        <!-- create aruco marker  -->
        <rosparam param="create_marker">false</rosparam>
        <!-- type of the aruco marker  -->
        <!-- 0: DICT_ARUCO_ORIGINAL 
            1 ~ 4: DICT_4X4_50, 100, 250, 1000
            5 ~ 8: DICT_5X5_50, 100, 250, 1000
            9 ~ 12: DICT_6X6_50, 100, 250, 1000
            13 ~ 16: DICT_7X7_50, 100, 250, 1000-->
        <rosparam param="marker_type">12</rosparam>
        <!-- set directory of the marker storage -->
        <rosparam param="store_directory">""</rosparam>
        <!-- set image conversion version
            "to_gray of to_rgb" -->
        <rosparam param="image_conversion_version">"to_rgb"</rosparam>
        <!-- camera matrix -->
        <rosparam param="fx">821.6405</rosparam>
        <rosparam param="fy">821.6648</rosparam>
        <rosparam param="cx">663.6956</rosparam>
        <rosparam param="cy">438.3716</rosparam>
        <!-- distortion coefficients  -->
        <rosparam param="k_">[-0.3794, 0.1686, 0.0, -0.3794, 0.1686, 0.0]</rosparam>
        <rosparam param="p_">[0.0, 0.0]</rosparam>
        <!-- length of the marker side -->
        <rosparam param="side_length">0.05</rosparam>
        <!-- visualize the marker detection result or not -->
        <rosparam param="vis_flag">false</rosparam>
    </node>
</launch>